test/functional/util/OrmUtils.ts

```diff
@@ -0,0 +1,106 @@
+import { expect } from "chai";
+import { OrmUtils } from "../../../src/util/OrmUtils";
+
+describe("OrmUtils.isPlainObject", () => {
+    const isPlainObject = OrmUtils.isPlainObject.bind(OrmUtils);
+
+    it("should return `true` if the Object is made by Object.create(null), {} or new Object().", () => {
+        class Foo {}
+
+        expect(isPlainObject(new Object())).to.be.true;
+        expect(isPlainObject(Object.create(null))).to.be.true;
+        expect(isPlainObject(Object.create(Object.prototype))).to.be.true;
+        expect(isPlainObject({ foo: "bar" })).to.be.true;
+        expect(isPlainObject({ constructor: "foo" })).to.be.true;
+        expect(isPlainObject({ constructor: Foo })).to.be.true;
+        expect(isPlainObject({}));
+    });
+
+    it("should return `false` if the Object is not made by Object.create(null), {} or new Object().", () => {
+        class Foo {}
+
+        expect(isPlainObject(Object.create({}))).to.be.false;
+        expect(isPlainObject(Object.create(Object))).to.be.false;
+        expect(isPlainObject(/foo/)).to.be.false;
+        expect(isPlainObject(function() {})).to.be.false;
+        expect(isPlainObject(() => {})).to.be.false;
+        expect(isPlainObject(["foo", "bar"])).to.be.false;
+        expect(isPlainObject([])).to.be.false;
+        expect(isPlainObject(new Foo())).to.be.false;
+        expect(isPlainObject(null)).to.be.false;
+        expect(isPlainObject(Math)).to.be.false;
+        expect(isPlainObject(Error)).to.be.false;
+        expect(isPlainObject(0)).to.be.false;
+        expect(isPlainObject(false)).to.be.false;
+        expect(isPlainObject(NaN)).to.be.false;
+    });
+ });
```

Comment:
since we're doing all the tests below - do we need to test `isPlainObject` if it's just an internal helper?  Do the below tests not go through all the possible permutations?